applications/nrf_desktop/src/modules/ble_conn_params.c

```diff
@@ -0,0 +1,294 @@
+		if (peers[i].conn == conn) {
+			return &peers[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int set_conn_params(struct bt_conn *conn, u16_t conn_latency,
+			   bool peer_llpm_support)
+{
+	int err;
+
+#ifdef CONFIG_BT_LL_NRFXLIB
+	if (peer_llpm_support) {
+		struct net_buf *buf;
+
+		hci_vs_cmd_conn_update_t *cmd_conn_update;
+
+		buf = bt_hci_cmd_create(HCI_VS_OPCODE_CMD_CONN_UPDATE,
+					sizeof(*cmd_conn_update));
+		if (!buf) {
+			LOG_ERR("Could not allocate command buffer");
+			return -ENOBUFS;
+		}
+
+		u16_t conn_handle;
+
+		err = bt_hci_get_conn_handle(conn, &conn_handle);
+		if (err) {
+			LOG_ERR("Failed obtaining conn_handle (err:%d)", err);
+			return err;
+		}
+
+		cmd_conn_update = net_buf_add(buf, sizeof(*cmd_conn_update));
+		cmd_conn_update->connection_handle   = conn_handle;
+		cmd_conn_update->conn_interval_us    = CONN_INTERVAL_LLPM_US;
+		cmd_conn_update->conn_latency        = conn_latency;
+		cmd_conn_update->supervision_timeout = CONN_SUPERVISION_TIMEOUT;
+
+		err = bt_hci_cmd_send_sync(HCI_VS_OPCODE_CMD_CONN_UPDATE, buf,
+					   NULL);
+	} else
+#endif /* CONFIG_BT_LL_NRFXLIB */
+	{
+		struct bt_le_conn_param param = {
+			.interval_min = CONN_INTERVAL_BLE_REG,
+			.interval_max = CONN_INTERVAL_BLE_REG,
+			.latency = conn_latency,
+			.timeout = CONN_SUPERVISION_TIMEOUT,
+		};
+
+		err = bt_conn_le_param_update(conn, &param);
+
+		if (err == -EALREADY) {
+			/* Connection parameters are already set. */
+			err = 0;
+		}
+	}
+
+	return err;
+}
+
+static int update_peer_conn_params(struct connected_peer *peer)
+{
+	__ASSERT_NO_MSG(peer);
+	struct bt_conn *conn = peer->conn;
+	u16_t latency = peer->requested_latency;
+
+	__ASSERT_NO_MSG(conn);
+	int err = set_conn_params(conn, latency, peer->llpm_support);
+
+	if (err) {
+		LOG_WRN("Cannot update conn parameters for peer %p (err:%d)",
+			peer, err);
+		/* Retry to update the connection parameters after an error. */
+		k_delayed_work_submit(&conn_params_update,
+				      CONN_PARAMS_ERROR_TIMEOUT);
+	} else {
+		LOG_INF("Conn params for peer: %p set: %s, latency: %"PRIu16,
+		  peer->conn,
+		  (IS_ENABLED(CONFIG_BT_LL_NRFXLIB) && peer->llpm_support) ?
+		  "LLPM" : "BLE", latency);
+	}
+
+	return err;
+}
+
+static bool conn_params_update_required(struct connected_peer *peer)
+{
+	if (!peer->conn) {
+		return false;
+	}
+
+	struct bt_conn_info info;
+	int err = bt_conn_get_info(peer->conn, &info);
+
+	if (err) {
+		LOG_ERR("Cannot get conn info (%d)", err);
+		return true;
+	}
+
+	__ASSERT_NO_MSG(info.role == BT_CONN_ROLE_MASTER);
+
+	if ((peer->llpm_support && (info.le.interval != CONN_INTERVAL_LLPM_REG)) ||
+	    (!peer->llpm_support && (info.le.interval != CONN_INTERVAL_BLE_REG)) ||
+	    (info.le.latency != peer->requested_latency)) {
+		return true;
+	}
+
+	return false;
+}
+
+static void conn_params_update_fn(struct k_work *work_desc)
+{
+	__ASSERT_NO_MSG(work_desc != NULL);
+
+	int err = 0;
+
+	for (size_t i = 0; (!err && (i < ARRAY_SIZE(peers))); i++) {
```

Comment:
err in on peer should not affect others

Reply:
The only error reason that I observed was that controller was busy. In that case we simply need to retry after some time.
I may remove this to have more generic approach (if controller is busy we will only have one additional failed try).